docs/decisions/0039-agent-interactions.md

```diff
@@ -0,0 +1,460 @@
+- Agent abstraction must adapt to different modalities
+- Agent of differing modality must be able to interact in a shared context. (One to many)
+- Agent of differing modality must be able to support their own modality requirements. (Specialization)
+- Must functionally support with patterns introduced by AutoGen
+- Must be extensible
+- Must encapsulate complexity within
+ implementation details, not calling patterns.
+
+> Note: Desire to establish AutoGen.Net alignment with SemanticKernel not limit to existing AutoGen definitions.
+
+### Agent Modalities:
+- **SemanticKernel - IChatCompletionService**: No storage (local history and agent definitions)
+- **SemanticKernel - ITextGenerationService**: Not needed / Out of scope (SLM?)
+- **Open AI Assistant API**: Remote storage managed by Assistant API (Open AI & Azure Open AI)
+- **AutoGen.Net**: Further specialization of the _Agent Architecture_
+
+## Design Concept
+
+Agents participate in a conversation, often in response to user or environmental input.  
+
+<img src="./diagrams/agent-analysis.png" alt="Agent Analysis Diagram" width="420" />
+
+In addition to `Agent`, two fundamental concepts are identified from this pattern:
+
+- Nexus: ("Conversation" from diagram) - Vehicle and context for a sequence of agent interactions.
+- Channel: ("Communication Path" from diagram) - The protocol with which the agent interacts with the nexus.
+
+> Agents of different modalities must be free to satisfy the requirements presented by their modality.  Formalizing the `Channel` concept provides natural vehicle for this to occur.
+
+These concepts come together to suggest the following generalization:
+
+<img src="./diagrams/agent-pattern.png" alt="Agent Pattern Diagram" width="212" />
+
+### Naming Alternatives
+
+Alternative names for *Nexus* and *Channel* concepts (feel free to come-up with your own):
+
+**Nexus**
+- Collection
+- Chat
+- Agency
+- Meeting
+- Team
+- Group
+
+**Channel**
+- Protocol
+- Connector
+- Connection
+- Conduit
+
+## Design Details
+
+<img src="./diagrams/agent-architecture.png" alt="Agent Architecture Diagram" width="700"  />
+
+### Class Dictionary
+
+Class Name|Parent Class|Role|Modality|Note
+-|-|-|-|-
+Agent|-|Agent|Abstraction|Root agent abstraction
+KernelAgent|Agent|Agent|Abstraction|Includes `Kernel` services and plug-ins
+NexusAgent|Agent|Agent|Utility|`Agent` adapter based on `AgentNexus`
+ProxyAgent|Agent|Agent|Abstraction|`Agent` adapter based on another (inner) `Agent`
+ChatCompletionAgent|KernelAgent|Agent|SemanticKernel|A functional agent based on `IChatCompletionService`
+GptAssistantAgent|KernelAgent|Agent|Assistant API|A functional agent based on *Open AI Assistant API*
+AgentChannel|-|Channel|Abstraction|Allows an agent to participate in a nexus.
+LocalChannel|AgentChannel|Channel|Utility|Concrete channel for agents based on local chat-history.
+GptAssistantChannel|AgentChannel|Channel|Assistant API|Channel associated with `GptAssistantAgent`
```

Comment:
What about `OpenAIAssistantChannel`?

Reply:
If Gemini is based on IChatCompletionService...then probably not. (did I mis-read the PR?)

Reply:
Not sure why Gemini is relevant here given this refers to the OpenAI Assistant API